src/charm.py

```diff
@@ -172,13 +215,25 @@ def _publish_sdcore_management_url(self, event: EventBase):
             management_url=self._get_webui_endpoint_url(),
         )
 
-    def _on_database_relation_broken(self, event: EventBase) -> None:
-        """Event handler for database relation broken.
+    def _on_common_database_relation_broken(self, event: RelationBrokenEvent) -> None:
+        """Event handler for common database relation broken.
 
         Args:
-            event: Juju event
+            event: Juju relation broken event
         """
-        self.unit.status = BlockedStatus("Waiting for database relation")
+        if not self.model.relations[COMMON_DATABASE_RELATION_NAME]:
```

Comment:
I am afraid this would not work on a broken relation event.
I think that the relation exists even if it is broken.
Maybe you can get the name of the relation from the event  and use it to get the status ?

Reply:
With the ops [2.10.0] (https://github.com/canonical/operator/releases/tag/2.10.0) by the help of that [PR](https://github.com/canonical/operator/pull/1091),  we are able to check the relations in that way. AMF operator also uses the same logic. (https://github.com/canonical/sdcore-amf-k8s-operator/blob/main/src/charm.py),